WordPress/gutenberg
# PreviewOptions: Fix critical error when children not passed

Fixes #54283

## What?

This PR fixes a critical error that occurs when the PreviewOptions component has no children.

## Why?

This is because this component expects to be passed a child that is a function. This component should not require children. Even the `PreviewOptions` component in the pattern-directory [only wants the device type dropdown](https://github.com/WordPress/pattern-directory/blob/d3c6c3618c50827d46e9101236984c06ebcd5419/public_html/wp-content/plugins/pattern-creator/src/components/header/index.js#L118).

## How?

Add `typof` check and Update document.

## Testing Instructions

Rewrite the Code block with a PreviewOptions component that has no children as below:

```javascript
/**
 * WordPress dependencies
 */
import {
	useBlockProps,
	__experimentalPreviewOptions as PreviewOptions,
} from '@wordpress/block-editor';
import { useDispatch, useSelect } from '@wordpress/data';

export default function CodeEdit() {
	const deviceType = useSelect( ( select ) => {
			return select( 'core/edit-post' ).__experimentalGetPreviewDeviceType();
	}, [] );
	const { __experimentalSetPreviewDeviceType: setPreviewDeviceType } =
		useDispatch( 'core/edit-post' );
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<PreviewOptions deviceType={ deviceType } setDeviceType={ setPreviewDeviceType } />
		</div>
	);
}
```

- Insert a Code block.
- trunk: The block should break.
- This PR: The dropdown should now work properly.

## Screenshots or screencast 

![preview-options](https://github.com/WordPress/gutenberg/assets/54422211/3367ce1d-8227-488c-966c-7d4f090f2b89)
